### PR TITLE
[tests] Fix running several test suites from the command line. Fixes xamarin/maccore@2279.

### PR DESCRIPTION
**Pull request by rolfbjarne**
Fixes https://github.com/xamarin/maccore/issues/2279

### CODE REVIEW

**Comment by xamarin-release-manager** [comment]
❌ [Build was (probably) aborted](https://jenkins.internalx.com/job/macios/job/PR-9348/1/display/redirect)

🔥 [Jenkins job](https://jenkins.internalx.com/job/macios/job/PR-9348/1/display/redirect) (on internal Jenkins) failed in stage(s) 'Test run' 🔥

✅ [Provisioning succeeded](https://jenkins.internalx.com/job/macios/job/PR-9348/1//console)
✅ [Build succeeded](https://jenkins.internalx.com/job/macios/job/PR-9348/1//console)
✅ Packages built successfully
<details><summary>View packages</summary>

* [xamarin.ios-13.21.1.460.pkg (Not notarized)](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9348/c046d5d17cbeae8b2863b5da38230aa0471e67e7/1/package/xamarin.ios-13.21.1.460.pkg)
* [xamarin.mac-6.21.1.460.pkg (Not notarized)](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9348/c046d5d17cbeae8b2863b5da38230aa0471e67e7/1/package/xamarin.mac-6.21.1.460.pkg)
* [Microsoft.iOS.Ref.13.6.100-ci.pr.gh9348.141+sha.c046d5d17.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9348/c046d5d17cbeae8b2863b5da38230aa0471e67e7/1/package/Microsoft.iOS.Ref.13.6.100-ci.pr.gh9348.141+sha.c046d5d17.nupkg)
* [Microsoft.iOS.Runtime.ios-arm.13.6.100-ci.pr.gh9348.141+sha.c046d5d17.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9348/c046d5d17cbeae8b2863b5da38230aa0471e67e7/1/package/Microsoft.iOS.Runtime.ios-arm.13.6.100-ci.pr.gh9348.141+sha.c046d5d17.nupkg)
* [Microsoft.iOS.Runtime.ios-arm64.13.6.100-ci.pr.gh9348.141+sha.c046d5d17.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9348/c046d5d17cbeae8b2863b5da38230aa0471e67e7/1/package/Microsoft.iOS.Runtime.ios-arm64.13.6.100-ci.pr.gh9348.141+sha.c046d5d17.nupkg)
* [Microsoft.iOS.Runtime.ios-x64.13.6.100-ci.pr.gh9348.141+sha.c046d5d17.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9348/c046d5d17cbeae8b2863b5da38230aa0471e67e7/1/package/Microsoft.iOS.Runtime.ios-x64.13.6.100-ci.pr.gh9348.141+sha.c046d5d17.nupkg)
* [Microsoft.iOS.Runtime.ios-x86.13.6.100-ci.pr.gh9348.141+sha.c046d5d17.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9348/c046d5d17cbeae8b2863b5da38230aa0471e67e7/1/package/Microsoft.iOS.Runtime.ios-x86.13.6.100-ci.pr.gh9348.141+sha.c046d5d17.nupkg)
* [Microsoft.iOS.Sdk.13.6.100-ci.pr.gh9348.141+sha.c046d5d17.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9348/c046d5d17cbeae8b2863b5da38230aa0471e67e7/1/package/Microsoft.iOS.Sdk.13.6.100-ci.pr.gh9348.141+sha.c046d5d17.nupkg)
* [Microsoft.macOS.Ref.10.15.100-ci.pr.gh9348.499+sha.c046d5d17.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9348/c046d5d17cbeae8b2863b5da38230aa0471e67e7/1/package/Microsoft.macOS.Ref.10.15.100-ci.pr.gh9348.499+sha.c046d5d17.nupkg)
* [Microsoft.macOS.Runtime.osx-x64.10.15.100-ci.pr.gh9348.499+sha.c046d5d17.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9348/c046d5d17cbeae8b2863b5da38230aa0471e67e7/1/package/Microsoft.macOS.Runtime.osx-x64.10.15.100-ci.pr.gh9348.499+sha.c046d5d17.nupkg)
* [Microsoft.macOS.Sdk.10.15.100-ci.pr.gh9348.499+sha.c046d5d17.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9348/c046d5d17cbeae8b2863b5da38230aa0471e67e7/1/package/Microsoft.macOS.Sdk.10.15.100-ci.pr.gh9348.499+sha.c046d5d17.nupkg)
* [Microsoft.tvOS.Ref.13.4.100-ci.pr.gh9348.499+sha.c046d5d17.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9348/c046d5d17cbeae8b2863b5da38230aa0471e67e7/1/package/Microsoft.tvOS.Ref.13.4.100-ci.pr.gh9348.499+sha.c046d5d17.nupkg)
* [Microsoft.tvOS.Runtime.tvos-arm64.13.4.100-ci.pr.gh9348.499+sha.c046d5d17.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9348/c046d5d17cbeae8b2863b5da38230aa0471e67e7/1/package/Microsoft.tvOS.Runtime.tvos-arm64.13.4.100-ci.pr.gh9348.499+sha.c046d5d17.nupkg)
* [Microsoft.tvOS.Runtime.tvos-x64.13.4.100-ci.pr.gh9348.499+sha.c046d5d17.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9348/c046d5d17cbeae8b2863b5da38230aa0471e67e7/1/package/Microsoft.tvOS.Runtime.tvos-x64.13.4.100-ci.pr.gh9348.499+sha.c046d5d17.nupkg)
* [Microsoft.tvOS.Sdk.13.4.100-ci.pr.gh9348.499+sha.c046d5d17.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9348/c046d5d17cbeae8b2863b5da38230aa0471e67e7/1/package/Microsoft.tvOS.Sdk.13.4.100-ci.pr.gh9348.499+sha.c046d5d17.nupkg)
* [Microsoft.watchOS.Ref.6.2.100-ci.pr.gh9348.499+sha.c046d5d17.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9348/c046d5d17cbeae8b2863b5da38230aa0471e67e7/1/package/Microsoft.watchOS.Ref.6.2.100-ci.pr.gh9348.499+sha.c046d5d17.nupkg)
* [Microsoft.watchOS.Runtime.watchos-arm.6.2.100-ci.pr.gh9348.499+sha.c046d5d17.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9348/c046d5d17cbeae8b2863b5da38230aa0471e67e7/1/package/Microsoft.watchOS.Runtime.watchos-arm.6.2.100-ci.pr.gh9348.499+sha.c046d5d17.nupkg)
* [Microsoft.watchOS.Runtime.watchos-x86.6.2.100-ci.pr.gh9348.499+sha.c046d5d17.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9348/c046d5d17cbeae8b2863b5da38230aa0471e67e7/1/package/Microsoft.watchOS.Runtime.watchos-x86.6.2.100-ci.pr.gh9348.499+sha.c046d5d17.nupkg)
* [Microsoft.watchOS.Sdk.6.2.100-ci.pr.gh9348.499+sha.c046d5d17.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9348/c046d5d17cbeae8b2863b5da38230aa0471e67e7/1/package/Microsoft.watchOS.Sdk.6.2.100-ci.pr.gh9348.499+sha.c046d5d17.nupkg)

</details>

🔥 [Failed to publish results](https://jenkins.internalx.com/job/macios/job/PR-9348/1//console) 🔥
✅ [API Diff (from stable)](http://xamarin-storage/jenkins/xamarin-macios/PR-9348/c046d5d17cbeae8b2863b5da38230aa0471e67e7/1/jenkins-results/api-diff/index.html)
🔥 [Failed to publish results](https://jenkins.internalx.com/job/macios/job/PR-9348/1//console) 🔥
✅ [API Diff (from PR only)](http://xamarin-storage/jenkins/xamarin-macios/PR-9348/c046d5d17cbeae8b2863b5da38230aa0471e67e7/1/jenkins-results/apicomparison/api-diff.html) (no change)
✅ [Generator Diff](http://xamarin-storage/jenkins/xamarin-macios/PR-9348/c046d5d17cbeae8b2863b5da38230aa0471e67e7/1/jenkins-results/generator-diff/index.html) (no change)
🔥 [Failed to publish results](https://jenkins.internalx.com/job/macios/job/PR-9348/1//console) 🔥
🔥 [Test run failed]() 🔥

File: tests/generator/Makefile [hunk 1]
```diff
@@ -10,7 +10,7 @@ build-unit-tests:
 
 run-unit-tests: build-unit-tests
 	rm -f .failed-stamp
-	$(SYSTEM_MONO) --debug $(XIBUILD_EXE_PATH) -t -- $(TOP)/packages/NUnit.ConsoleRunner.3.9.0/tools/nunit3-console.exe $(abspath $(TOP)/tests/generator/bin/Debug/generator-tests.dll) "--result=$(abspath $(CURDIR)/TestResult.xml);format=nunit2" $(TEST_FIXTURE) --labels=All || touch $(CURDIR)/.failed-stamp
+	$(TOP)/tools/nunit3-console-3.11.1 $(abspath $(TOP)/tests/generator/bin/Debug/generator-tests.dll) "--result=$(abspath $(CURDIR)/TestResult.xml);format=nunit2" $(TEST_FIXTURE) --labels=All || touch $(CURDIR)/.failed-stamp
```
Comment (spouliot) [inline tests/generator/Makefile:13]: Seems easy to miss one/some
Maybe make `$(TOP)/tools/nunit3-console-3.11.1` a variable that is reused everywhere its needed ?

Comment (rolfbjarne) [reply tests/generator/Makefile:13]: It depends on the NUnitLite version defined in the csproj for each test project, which complicates matters.

A thought I had (just after implementing this fix) was to make a generic `$(TOP)/tools/nunit3-console` script that takes the csproj as an argument, and parses it to determine what to do. In the end I decided to postpone this until the next time the problem shows up.

**Comment by rolfbjarne** [comment]
Test failures are unrelated

* monotouch-test/iOS 64-bits: https://github.com/xamarin/maccore/issues/2165
* monotouch-test/watchOS 32-bits: https://github.com/xamarin/maccore/issues/581
* Sample test failures: https://github.com/xamarin/maccore/issues/1124

**Pull request merged**